src/python-frontend/python_converter.cpp

```diff
@@ -337,42 +313,75 @@ void python_converter::get_if_statement(
   then.location() = location;
 
   // Create if code and append "then" block
-  codet code_if("ifthenelse");
-  code_if.copy_to_operands(cond, convert_expression_to_code(then));
+  codet code;
+  auto type = ast_node["_type"];
+  if(type == "If")
+    code.set_statement("ifthenelse");
+  else if(type == "While")
+    code.set_statement("while");
+
+  code.copy_to_operands(cond, convert_expression_to_code(then));
 
   // Append 'else' block to the statement
   if(ast_node.contains("orelse") && !ast_node["orelse"].empty())
   {
     exprt else_expr = get_block(ast_node["orelse"]);
-    code_if.copy_to_operands(convert_expression_to_code(else_expr));
+    code.copy_to_operands(convert_expression_to_code(else_expr));
   }
 
-  target_block.copy_to_operands(code_if);
+  target_block.copy_to_operands(code);
 }
 
-bool python_converter::convert()
+exprt python_converter::get_block(const nlohmann::json &ast_block)
 {
-  codet main_code = code_blockt();
-  main_code.make_block();
-  current_function_name = "globalscope";
+  code_blockt block;
 
-  for(auto &element : ast_json["body"])
+  // Iterate through the statements of the block
+  for(auto &element : ast_block)
   {
     StatementType type = get_statement_type(element);
 
-    // Variable assignments
-    if(type == StatementType::VARIABLE_ASSIGN)
+    switch(type)
+    {
+    case StatementType::VARIABLE_ASSIGN:
     {
-      get_var_assign(element, main_code);
+      // Add an assignment to the block
+      get_var_assign(element, block);
+      break;
     }
-    // If statements
-    else if(type == StatementType::IF_STATEMENT)
+    case StatementType::IF_STATEMENT:
+    case StatementType::WHILE_STATEMENT:
     {
-      get_if_statement(element, main_code);
+      get_conditional_stms(element, block);
+      break;
+    }
+    case StatementType::COMPOUND_ASSIGN:
+    {
+      get_compound_assign(element, block);
+      break;
+    }
+    case StatementType::FUNC_DEFINITION:
+    case StatementType::UNKNOWN:
+    default:
+      log_error("error");
+      abort();
     }
   }
 
-  // add "main"
+  return block;
+}
+
+bool python_converter::convert()
+{
+  current_function_name = "globalscope";
```

Comment:
I wonder what this string is being used for. Given that `globalscope` could well be a valid Python function's name it seems confusing.